build-system/tasks/serve.js

```diff
@@ -81,15 +113,26 @@ function getMiddleware() {
 /**
  * Launches a server and waits for it to fully start up
  *
- * @param {?Object} connectOptions
- * @param {?Object} serverOptions
- * @param {?Object} modeOptions
+ * @param {?ConnectOptionsDef} connectOptions
+ * @param {?ServerOptionsDef} serverOptions
+ * @param {?ModeOptionsDef} modeOptions
  */
 async function startServer(
   connectOptions = {},
   serverOptions = {},
   modeOptions = {}
 ) {
+  if (
+    'new_server' in serverOptions ? serverOptions.new_server : argv.new_server
+  ) {
+    try {
+      buildNewServer();
+    } catch {
+      log(red('ERROR:'), 'Could not build', cyan('AMP Dev Server'));
+      return;
+    }
+  }
+
```

Comment:
Why was this moved here? `doServe()` already does this during normal startup.

Reply:
sorry was just testing some stuff out originally on why it wasnt logging "serving esm", will revert.